### DISCUSSION TITLE
CSS Styling Not Displaying Correctly v3

### DISCUSSION THREAD

**Discussion by itsthecheat** [gatsbyjs/gatsby#32448] [Help]
Hello,

I've recently migrated to v3, but I haven't been able to figure out why all the styling isn't showing up correctly. The first screenshot is just an example of one of the pages and how it should look; that is what is currently in production at lesliethe.dev prior to changing some things after moving to v3. The bottom is how it currently looks in Development (https://github.com/itsthecheat/dev-portfolio). There is a source map error, is that the issue? I've updated plugins, and currently import bootstrap in gatsby-browser, and use modules for the components. Thanks in advance for any input!

![Screen Shot 2021-07-20 at 4 02 37 PM](https://user-images.githubusercontent.com/17536124/126388452-dfe38974-25d2-4977-a6d8-a2ea5e1845d4.png)
![Screen Shot 2021-07-20 at 4 02 26 PM](https://user-images.githubusercontent.com/17536124/126388462-75e01a6b-c6aa-4d1b-a86d-04d8290e2cf5.png)

**Comment by LpmRaven** [comment]
Can you show me any css errors that occur when you build? It looks like you are using REMs for the margin between form items, I would recommend you give the component containing them a `font-size: 1rem` and then use `em` to set the margin. For more about that, check out [css-tricks explaination of rems and ems](https://css-tricks.com/rems-ems/). Secondly, as you are using the large light/sun as a background, I would suggest you explicitly give it a `z-index: -1;`.

**Reply by itsthecheat** [reply]
These are the errors in development:
![Screen Shot 2021-07-21 at 1 16 26 PM](https://user-images.githubusercontent.com/17536124/126531490-1e2e10d8-ad8c-4e4f-b278-54ccc0abbb2e.png)

And then after build with gatsby serve
![Screen Shot 2021-07-21 at 1 22 18 PM](https://user-images.githubusercontent.com/17536124/126532340-f3f56c98-9e9f-4d84-bc88-e5f555eeb42e.png)
:
For some reason certain styling is not being applied to the entire site, another example of the about page before (how it should look) and after:
![Screen Shot 2021-07-21 at 1 25 00 PM](https://user-images.githubusercontent.com/17536124/126532820-c284f70e-3e5f-4ef9-856c-1203719a821b.png)
![Screen Shot 2021-07-21 at 1 25 07 PM](https://user-images.githubusercontent.com/17536124/126532842-b98d54d3-7a52-41c1-900e-393ac023a925.png)

**Reply by LpmRaven** [reply]
Could you paste me your gatsby-config and the gatsby build from the terminal?

**Reply by itsthecheat** [reply]
**Here's the gatsby build:**

verbose set gatsby_log_level: "verbose"
verbose set gatsby_executing_command: "build"
verbose loading local command from:
/Users/lesliebehum/Development/dev-portfolio/node_modules/gatsby/dist/commands/build.js
verbose running command: build

success open and validate gatsby-configs, load plugins - 1.385s
success onPreInit - 0.031s
success initialize cache - 0.005s
success copy gatsby files - 0.134s
verbose Attaching functions to development server
success Compiling Gatsby Functions - 0.640s
success onPreBootstrap - 0.685s
success createSchemaCustomization - 0.017s
verbose Checking for deleted pages
verbose Deleted 0 pages
verbose Found 0 changed pages
success Checking for changed pages - 0.004s
success source and transform nodes - 0.349s
success building schema - 0.762s
success createPages - 0.050s
success createPagesStatefully - 0.076s
info Total nodes: 107, SitePage nodes: 12 (use --verbose for breakdown)
verbose Number of node types: 9. Nodes per type: Directory: 2, File: 23, ImageSharp: 21, MarkdownRemark: 1,
ProjectsYaml: 6, Site: 1, SiteBuildMetadata: 1, SitePage: 12, SitePlugin: 40
verbose Checking for deleted pages
verbose Deleted 0 pages
verbose Found 0 changed pages
success Checking for changed pages - 0.005s
success Cleaning up stale page-data - 0.003s
success update schema - 0.106s
success onPreExtractQueries - 0.004s
success extract queries from components - 4.654s
success write out redirect data - 0.002s
warn The icon(./src/images/logo_slash.png) you provided to 'gatsby-plugin-manifest' is not square.
The icons we generate will be square and for the best results we recommend you provide a square icon.

success Build manifest and related icons - 0.085s
success onPostBootstrap - 0.098s
info bootstrap finished - 14.124s
success write out requires - 0.005s
success Building production JavaScript and CSS bundles - 1.494s
success Caching JavaScript and CSS webpack compilation - 0.006s
success Rewriting compilation hashes - 0.007s
success Writing page-data.json files to public directory - 0.002s - 0/0 0.00/s
success Building HTML renderer - 1.365s
success Caching HTML renderer compilation - 0.489s
success Building static HTML for pages - 0.866s - 12/12 13.86/s
success onPostBuild - 0.001s
info Done building in 18.217995691 sec

**Here's the gatsby-config:**

module.exports = {
  plugins: [
   {
     resolve: `gatsby-plugin-layout`,
     options: {
       component: require.resolve(`./src/components/Layout`),
     },
   },
   {
    resolve: `gatsby-plugin-manifest`,
    options: {
      name: "Leslie Behum",
      short_name: "LB",
      start_url: "/",
      background_color: "#000000",
      theme_color: "#000000",
      // Enables "Add to Homescreen" prompt and disables browser UI (including back button)
      // see https://developers.google.com/web/fundamentals/web-app-manifest/#display
      display: "standalone",
      icon: "./src/images/logo_slash.png", // This path is relative to the root of the site.
      // An optional attribute which provides support for CORS check.
      // If you do not provide a crossOrigin option, it will skip CORS for manifest.
      // Any invalid keyword or empty string defaults to `anonymous`
      crossOrigin: `use-credentials`,
  },
    },
  `gatsby-plugin-postcss`,
  `gatsby-plugin-preload-fonts`,
   `gatsby-plugin-image`,
   `gatsby-transformer-sharp`,
   `gatsby-plugin-sharp`,
   `gatsby-transformer-remark`,
   `gatsby-transformer-yaml`,
   `gatsby-plugin-react-helmet`,
   {
  resolve: `gatsby-source-filesystem`,
  options: {
    name: `data`,
    path: `${__dirname}/src/content`,
  },
},
{
resolve: `gatsby-source-filesystem`,
options: {
 name: `img`,
 path: `${__dirname}/src/images`,
},
},
 ],
 siteMetadata: {
  title: `Leslie Behum | Web Developer in NYC`,
  siteUrl: `http://www.lesliethe.dev`,
  description: `Web Developer based in Brooklyn, NY`,
},
}

**Comment by graysonhicks** [comment]
Hi @itsthecheat do you have the v3 version deployed anywhere?

**Comment by itsthecheat** [comment]
Hello,
I haven’t deployed it yet

**Comment by amykapernick** [comment]
I'm having the same issue, it works fine on dev but as soon as I build it, it's missing part of the styles - [Netlify Staging](https://60f90a80d6515d000831128f--aimhigher.netlify.app)

For example, this is the styles that should be on the footer at 370px wide, and what is actually there
![image](https://user-images.githubusercontent.com/15953185/126596651-876381a5-cfde-46c1-adaf-dd691c3f2bc2.png)
![image](https://user-images.githubusercontent.com/15953185/126596720-450a98d4-2f7c-4d1e-bddf-9d826f53192a.png)

There seems to be something where the inital styles aren't there, but any nested sass styles do come through, and anything in a media query, and only on some components

```
.footer {
	// These styles are not there

	&:before, &:after {
		// These styles are there
	}

	&:before {
		// These styles are there
	}

	&:after {
		// These styles are there
	}
}

@media(min-width: 40em) {
	.footer {
		// These styles are there
	}
}
```

**Comment by itsthecheat** [comment]
Hello all,

So after just now looking into the source map issue a bit further, I think I finally figured it out. When I upgraded to v3 I went ahead and updated all the packages in package.json as well,  including Bootstrap. I also use react-bootstrap, so after going back to the react-bootstrap installation to check it out, I'm realizing it only works with up to Bootstrap 4.6.0. I went back to that version of Bootstrap and the styling is back the way it should be.

I'm still having a few minor issues seemingly unrelated to the other problem.

Thanks for taking the time to look into it, much appreciated.
